### PR TITLE
MongoDB Adapter Batch_Create and Pagination in find()

### PR DESCRIPTION
**Pull request by abdullah-syngenta**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ boto3 = "*"
 elasticsearch = "==7.13.4"
 jsonref = "*"
 jsonpickle = "*"
-pymongo= "*"
+pymongo= "==4.2.0"
```
Comment (paulcruse-syn) [inline Pipfile:15]: this should stay as a `*`... the applications using this package should be the ones to limit the version, except in extreme cases like `elasticsearch`

```suggestion
pymongo= "*"
```

File: syngenta_digital_dta/mongo/adapter.py [hunk 2]
```diff
@@ -46,6 +46,16 @@ def create(self, **kwargs):
...
         return data
 
+    def batch_create(self, **kwargs):
+        items = []
+        for item in kwargs['data']:
+            item = schema_mapper.map_to_schema(item, self.model_schema_file, self.model_schema)
+            item['_id'] = item[self.model_identifier]
+            items.append(item)
+        self.connection.insert_many(items)
+        super().publish('create', items, **kwargs)
```
Comment (paulcruse-syn) [inline syngenta_digital_dta/mongo/adapter.py:56]: Downstream users will be need to know that this is sending a list of items

```suggestion
        super().publish('batch_create', items, **kwargs)
```

File: syngenta_digital_dta/mongo/adapter.py [hunk 3]
```diff
@@ -61,7 +71,12 @@ def find_one(self, **kwargs):
         return self.connection.find_one(kwargs['query'])
 
     def find(self, **kwargs):
-        results = self.connection.find(kwargs['query'])
+        if kwargs.get('page_size') and kwargs.get('page_number'):
+            skips = kwargs['page_size'] * (kwargs['page_number'] - 1)
+            results = self.connection.find(kwargs['query']).skip(skips).limit(kwargs['page_size'])
+        else:
+            results = self.connection.find(kwargs['query'])
+
```
Comment (paulcruse-syn) [inline syngenta_digital_dta/mongo/adapter.py:79]: This isn't very readable and doesn't support the other optional parameters possible, so it's not very extensible:

```suggestion
        kwargs.pop('operation', None)
        query = kwargs.pop('query')
        results = self.connection.find(query, **kwargs)
```

**Review by paulcruse-syn — CHANGES_REQUESTED** [review]
Some things we need to change to make it more readable and extensible.

File: syngenta_digital_dta/mongo/adapter.py (hunk 3 above)
Comment (abdullah-syngenta) [reply syngenta_digital_dta/mongo/adapter.py]: this is much better

File: syngenta_digital_dta/mongo/adapter.py (hunk 2 above)
Comment (abdullah-syngenta) [reply syngenta_digital_dta/mongo/adapter.py]: Sure

File: Pipfile (hunk 1 above)
Comment (abdullah-syngenta) [reply Pipfile]: No problem

**Review by paulcruse-syn — APPROVED** [review]

**Pull request merged**